modules/cudev/include/opencv2/cudev/grid/reduce_to_vec.hpp

```diff
@@ -189,6 +189,7 @@ __host__ void gridReduceToColumn_(const SrcPtr& src, GpuMat_<ResType>& dst, cons
                                                                 shrinkPtr(mask),
                                                                 rows, cols,
                                                                 StreamAccessor::getStream(stream));
+    dst = dst.reshape(dst.channels(), rows);
```

Comment:
Please explicitly cast the `reshape` result to `GpuMat_`:

```
dst = GpuMat_<ResType>(dst.reshape(dst.channels(), rows));
```

Otherwise compilation fails.